src/test/java/org/opentripplanner/middleware/ApiKeyManagementTest.java

```diff
@@ -22,32 +22,33 @@
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 import static org.opentripplanner.middleware.TestUtils.getBooleanEnvVar;
 import static org.opentripplanner.middleware.TestUtils.mockAuthenticatedRequest;
-import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.auth.Auth0Connection.authDisabled;
+import static org.opentripplanner.middleware.controllers.api.ApiUserController.DEFAULT_USAGE_PLAN_ID;
 
 /**
- * Tests for creating and deleting api keys. The following config parameters are must be set in
+ * Tests for creating and deleting api keys. The following config parameters must be set in
  * configurations/default/env.yml for these end-to-end tests to run:
- *  - RUN_E2E=true the end-to-end environment variable must be set (NOTE: this is not a config value)
- *  - An AWS_PROFILE is required, or AWS access has been configured for your operating environment e.g.
- *    C:\Users\<username>\.aws\credentials in Windows or Mac OS equivalent.
- *  - DISABLE_AUTH set to true to bypass auth checks and use users defined here. DEFAULT_USAGE_PLAN_ID set to a valid usage
- *    plan id. AWS requires this to create an api key.
- *  - TODO: It might be useful to allow this to run without DISABLE_AUTH set to true (in an end-to-end environment using
- *      real tokens from Auth0.
+ * - RUN_E2E=true the end-to-end environment variable must be set (NOTE: this is not a config value)
+ * - An AWS_PROFILE is required, or AWS access has been configured for your operating environment. E.g.,
+ *      C:\Users\<username>\.aws\credentials in Windows or Mac OS equivalent.
+ * - DISABLE_AUTH set to true to bypass auth checks and use users defined here.
+ * - DEFAULT_USAGE_PLAN_ID set to a valid usage plan id. AWS requires this to create an api key.
  */
 public class ApiKeyManagementTest extends OtpMiddlewareTest {
     private static final Logger LOG = LoggerFactory.getLogger(ApiKeyManagementTest.class);
-    private static String DEFAULT_USAGE_PLAN_ID;
     private static ApiUser apiUser;
     private static AdminUser adminUser;
+    // TODO: It might be useful to allow this to run without DISABLE_AUTH set to true (in an end-to-end environment
+    //  using real tokens from Auth0.
+    private static boolean testsShouldRun = getBooleanEnvVar("RUN_E2E") && authDisabled();
 
     /**
      * Create an {@link ApiUser} and an {@link AdminUser} prior to unit tests
      */
     @BeforeAll
     public static void setUp() throws IOException {
+        assumeTrue(testsShouldRun);
```

Comment:
This is never returning true because the config has yet to be loaded, causing the authDisabled call to always return false.

Reply:
Wait nvm. This wasn't working because I was running the tests in just this class isolated from the rest of the test suite. Regardless, a good general principle of testing (I know we don't adhere to this with the datatools e2e test) is that any test should be able to run independent of others and it might be a good idea to refactor things a bit so it works here.

Reply:
Good catch. Fixed in 69c562fa9223d09d1fde73eac6f92de28d58d3d4